### PR TITLE
table: check valid UTF8 value for UTF8 column

### PR DESCRIPTION
**Pull request by coocood**
We need to check if the input data is valid UTF8 string if the column charset is UTF8,
Otherwise, it may cause more serious error in the future, and hard to fix.

This prevents future issue like https://github.com/pingcap/tidb/issues/1808

### CODE REVIEW

**Comment by coocood** [comment]
@shenli Yes when we insert or update utf8 column, but that's something we have to pay.
I checked out the Go implements, which check the string one byte at a time. Rust implementation checks string one word(64bit) at a time, which should be more efficient, we can optimize it if that becomes a bottleneck.

`utf8.Valid` adds about 2 ns for a byte.

**Comment by shenli** [comment]
LGTM

File: table/tables/tables.go [hunk 1]
```diff
@@ -212,6 +213,9 @@ func (t *Table) UpdateRecord(ctx context.Context, h int64, oldData []types.Datum
 			}
 			currentData[i] = defaultVal
 		}
+		if col.Charset == "utf8" && !utf8.Valid(currentData[i].GetBytes()) {
```
Comment (shenli) [inline table/tables/tables.go]: Should we also check col.Tp?

Comment (coocood) [inline table/tables/tables.go]: Only String types use 'utf8' charset.

**Comment by coocood** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**